### DISCUSSION TITLE
What's the purpose of blocking with Updater class's idle()?

### DISCUSSION THREAD

**Discussion by samuelwhiskeyjohnson** [python-telegram-bot/python-telegram-bot#3947] [Q&A]
What's the purpose of blocking with Updater class's idle()?
When I put idle() the program doesn't end.
![image](https://github.com/python-telegram-bot/python-telegram-bot/assets/71688370/fb9ab5eb-c64e-440a-abe0-deb0880c3049)

When I comment it out, the program execution ends.
![image](https://github.com/python-telegram-bot/python-telegram-bot/assets/71688370/daa46739-525b-4dfe-b7c1-7f05beebbae6)

But either way, the bot still works.

``` py
#!/usr/bin/env python3

from telegram.ext import Updater, CommandHandler


def start_handler(update, context):
    context.bot.send_message(chat_id=update.effective_chat.id,
                             text="Hello from python!")
    
    
# open() "r" means read
with open("/home/pi/Desktop/telegramBotToken", "r") as tokenFile:
    telegramToken = tokenFile.read().rstrip()
    
updater = Updater(token=telegramToken)
dispatcher = updater.dispatcher
dispatcher.add_handler(CommandHandler("start", start_handler))

# starts receiving message indefintely
updater.start_polling()
print("program is running")

# I can exit with ctrl+C or other terminational signal
updater.idle() #whether you comment this line or not, bot still works.
print("program ended")


```

**Comment by Bibo-Joshi** [comment]
Hi. We no longer provide support for PTB versions older than 20.x. please upgrade to the latest version. There is a transition guide in the wiki.
